model/generate_test.go

```diff
@@ -1182,6 +1251,48 @@ func (s *GenerateSuite) TestSaveNewTasksWithDependencies() {
 	s.Equal(4, generatorTask.NumGeneratedTasks)
 }
 
+func (s *GenerateSuite) TestSaveNewTasksWithDependenciesInNewBuilds() {
+	generator := task.Task{
+		Id:          "task_that_called_generate_task",
+		DisplayName: "task_that_called_generate_task",
+		Version:     "version_that_called_generate_task",
+		BuildId:     "sample_build",
+		Activated:   true,
+	}
+	s.NoError(generator.Insert())
+	sampleBuild := build.Build{
+		Id:           "sample_build",
+		BuildVariant: "a_variant",
+		Version:      "version_that_called_generate_task",
+	}
+	v := &Version{
+		Id:            "version_that_called_generate_task",
+		BuildIds:      []string{"sample_build"},
+		BuildVariants: []VersionBuildStatus{},
+	}
+	pp := &ParserProject{}
+	err := util.UnmarshalYAMLWithFallback([]byte(projYmlTwoBVs), &pp)
+	s.NoError(err)
+	pp.Id = "version_that_called_generate_task"
+	s.NoError(pp.Insert())
+	s.NoError(sampleBuild.Insert())
+	s.NoError(v.Insert())
+
+	g := generatedProjectWithNewBuild
+	g.Task = &generator
+	p, pp, err := FindAndTranslateProjectForVersion(s.ctx, s.env.Settings(), v, false)
+	s.Require().NoError(err)
+	p, pp, v, err = g.NewVersion(context.Background(), p, pp, v)
+	s.NoError(err)
+	s.NoError(g.Save(s.ctx, s.env.Settings(), p, pp, v))
+
+	tasks := []task.Task{}
+	s.NoError(db.FindAllQ(task.Collection, db.Query(bson.M{task.VersionKey: v.Id}), &tasks))
+	s.Require().Len(tasks, 4)
+	for _, t := range tasks {
```

Comment:
So does this fail without these changes because some task is newly added, but was marked as activate false?

Reply:
yes, the dependency is activate:false and in a newly added build!